packages/flutter_tools/lib/src/ios/core_devices.dart

```diff
@@ -0,0 +1,847 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/logger.dart';
+import '../base/process.dart';
+import '../convert.dart';
+import '../device.dart';
+import '../macos/xcode.dart';
+
+/// A wrapper around the `devicectl` command line tool.
+///
+/// CoreDevice is a device connectivity stack introduced in Xcode 15. Devices
+/// with iOS 17 or greater are CoreDevices.
+///
+/// `devicectl` (CoreDevice Device Control) is an Xcode CLI tool used for
+/// interacting with CoreDevices.
+class IOSCoreDeviceControl {
+  IOSCoreDeviceControl({
+    required Logger logger,
+    required ProcessManager processManager,
+    required Xcode xcode,
+    required FileSystem fileSystem,
+  })  : _logger = logger,
+        _processUtils = ProcessUtils(logger: logger, processManager: processManager),
+        _xcode = xcode,
+        _fileSystem = fileSystem;
+
+  final Logger _logger;
+  final ProcessUtils _processUtils;
+  final Xcode _xcode;
+  final FileSystem _fileSystem;
+
+  /// When the `--timeout` flag is used with `devicectl`, it must be at
+  /// least 5 seconds. If lower than 5 seconds, `devicectl` will error and not
+  /// run the command.
+  static const int _minimumTimeoutInSeconds = 5;
+
+  /// Executes `devicectl` command to get list of devices. The command will
+  /// likely complete before [timeout] is reached. If [timeout] is reached,
+  /// the command will be stopped as a failure.
+  Future<List<Object?>> _listCoreDevices({
+    Duration timeout = const Duration(seconds: _minimumTimeoutInSeconds),
+  }) async {
+    if (!_xcode.isDevicectlInstalled) {
+      _logger.printError('devicectl is not installed.');
+      return <Object?>[];
+    }
+
+    // Default to minimum timeout if needed to prevent error.
+    Duration validTimeout = timeout;
+    if (timeout.inSeconds < _minimumTimeoutInSeconds) {
+      _logger.printError(
+          'Timeout of ${timeout.inSeconds} seconds is below the minimum timeout value '
+          'for devicectl. Changing the timeout to the minimum value of $_minimumTimeoutInSeconds.');
+      validTimeout = const Duration(seconds: _minimumTimeoutInSeconds);
+    }
+
+    final Directory tempDirectory = _fileSystem.systemTempDirectory.createTempSync('core_devices.');
+    final File output = tempDirectory.childFile('core_device_list.json');
+    output.createSync();
+
+    final List<String> command = <String>[
+      ..._xcode.xcrunCommand(),
+      'devicectl',
+      'list',
```

Comment:
will `xcrun devicectl list` only return ios 17 devices?

Reply:
As it currently is, yes